### ISSUE TITLE
The error  Viscosity is zero

### ISSUE THREAD

**Issue by tyszwh** [underworldcode/underworld1#2]
Dear all
what means with this error message in underworld1.
It occurs when the model run some times.
Stgermain-error -----Viscosity is zero
Is some parameter set wrong in my model?

**Comment by jmansour** [comment]
You haven't really given us much to work with, but it appears that, as per the message, somewhere in your domain the viscosity is zero.  Perhaps try using the ViscosityField component to visualise the viscosity.

**Comment by tyszwh** [comment]
Dear  jmansour 
I try to make a rift model and obtain the vertical displacement data.
Does the Y axis always set as a vertical axis in the underworld? and what should I do to obtain the ascii file of vertical displacement data which produces by model?
it seems like the problem occurs when I try to set the z axis as vertical axis.

**Comment by jmansour** [comment]
Generally in Underworld1, yes the y axis is set as the vertical axis.  

Can I ask why you are using Underworld1 for this problem?  Your previous github questions related to Underworld2.

**Comment by tyszwh** [comment]
Yes , I also compiled the Underworld2 in my machine. But I have not migrated fully from underworld1 to underworld2 due to I am not very familiar with Underwolrd2 now. So I still use underworld1 when I hope obtain some data  which produce by Underworld for other model input. Could you give me some suggestion for this problem? Should I try to use Underworld2 for obtain these data and fully abandoned the underworld1 ?

**Comment by jmansour** [comment]
I would generally advise users to switch to UW2 as soon as possible.  It is much easier for you to debug your issues in UW2, and likewise it's easier for us to help you with any issues you encounter using UW2. 

However, not all functionality provided by UW1 is available in UW2 (though the reverse is more so true), so it depends on what functionality your UW1 models use.  

I think you would be best trying to construct your models in UW2, and see if you hit any roadblocks. 

For your current problems, if you can attach/send a simplified version which reproduces the issue, I can try and have a look, but I really can't spend a lot of time on it.  By simplified, I mean it should run well on 1 processor, and have *all* unnecessary details removed.

**Comment by tyszwh** [comment]
Dear  jmansour 
Ok， Thanks a lot for you advise. I  will try to construct model in UW2 firstly, and then reorganize the problem if I meet too. 
What parameter should I set to direct obtain the  ASCII file of vertical displacement data.

**Comment by jmansour** [comment]
Can you explain specifically what you mean by vertical displacement data?

**Comment by tyszwh** [comment]
Dear jmansour
The middle part of box model should be sink obviously when the Velocity boundary act on the left part and right part of model.  I hope obtain the these displacement data(all the model position ) in vertical direction at different time step.

**Comment by jmansour** [comment]
I'm not exactly sure, but it sounds like you will want to use passive tracers to measure your displacements.  This can be done using UW1 or UW2, but it certainly will be easier using using UW2.

Basically, you will add a line of particles where you want to measure the displacement, and then advect the particles, possibly using only the vertical component of velocity.  From this you can calculate your displacement.

**Comment by tyszwh** [comment]
Thanks, It is really helpful.
